tests/nlu/featurizers/test_regex_featurizer.py

```diff
@@ -523,24 +488,17 @@ def test_persist_load_for_finetuning(tmp_path: Path):
 
     # Test merging of a new pattern to an already trained component.
     assert len(loaded_featurizer.known_patterns) == 4
-    assert loaded_featurizer.vocabulary_stats == {
-        "max_number_patterns": 8,
-        "pattern_slots_filled": 4,
-    }
 
 
-def test_incremental_train_featurization(tmp_path: Path):
+def test_finetuning(tmp_path: Path):
```

Comment:
Naming this function to something like `test_vocabulary_expand_for_finetuning` will be better.